module/zfs/spa.c

```diff
@@ -2909,6 +2990,36 @@ spa_load_impl(spa_t *spa, uint64_t pool_guid, nvlist_t *config,
 		spa_activate(spa, orig_mode);
 
 		return (spa_load(spa, state, SPA_IMPORT_EXISTING, B_TRUE));
+
+	} else if (!nvlist_exists(config, ZPOOL_CONFIG_HAS_PER_VDEV_ZAPS)) {
+		/*
+		 * The current config is missing vdev zaps. The top vdev
+		 * zap is required before vdev metaslabs are initialized
+		 * for both dRAID and the special allocation class.
```

Comment:
can we nuke all of this once @pzakha's changes to always trust the mos config land in ZoL?
(also, we probably want the comment to not mention dRAID yet)

Reply:
Yeah definitely. In the new logic we do not transfer stuff from MOS-config into user-config.
Instead we transfer the vdev paths from user-config into MOS-config and discard user-config.

Reply:
Looking forward to picking up the refactored spa load and this work-around can be removed!